src/OidcClientInterface.php

```diff
@@ -75,4 +75,12 @@ public function generateEndSessionEndpointRedirect(
    * @throws OidcException
    */
   public function retrieveUserInfo(OidcTokens $tokens): OidcUserData;
+
+  /**
+   * Retrieve additional information from the idtoken.
+   *
+   * @throws OidcException
+   */
+  public function retrieveDataFromIdToken(OidcTokens $tokens): array;
```

Comment:
Adding a method to the interface would be a BC break for people implementing this interface, so I am a bit hesitant about this.